llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -13597,6 +13599,13 @@ struct NodeExtensionHelper {
     case RISCVISD::VZEXT_VL:
     case RISCVISD::FP_EXTEND_VL:
       return OrigOperand.getOperand(0);
+    case ISD::SPLAT_VECTOR: {
+      SDValue Op = OrigOperand.getOperand(0);
+      if (Op.getOpcode() == ISD::FP_EXTEND)
+        return Op;
```

Comment:
How does this work for fixed length float splats currently if we didn't already handle `RISCVISD::VFMV_V_F_VL`? Since in #81248 it looks like we emit vfwadd.vf

Reply:
Why do we need to look through scalar FP_EXTEND here, but we don't need to look throug ZERO_EXTEND or SIGN_EXTEND for integer?

Reply:
> >How does this work for fixed length float splats currently if we didn't already handle RISCVISD::VFMV_V_F_VL? Since in https://github.com/llvm/llvm-project/pull/81248 it looks like we emit vfwadd.vf

> The extend on those tests are happening in the vector domain not the scalar domain I think. 

Yes, you are right. We also need to handle the extension in the scalar domain (i.e. similar to  #87249) for `RISCVISD::VFMV_V_F_VL`. Thanks for pointing out this.

Reply:
> Why do we need to look through scalar FP_EXTEND here, but we don't need to look throug ZERO_EXTEND or SIGN_EXTEND for integer? 

For integers, I think the `DAG.getSplat` would be an implicit truncation for us. But it seems that the float does not work.